[MSVC] 'sscanf()' and "%m"

### Feature Description

Building **gnuradio-qtgui.dll** on MSVC (with OpenGL support), I noted this important warning:
```c
QFosphorColorMapper.cc(67): warning C4476: 'sscanf' : unknown type field character 'm' in format specifier
QFosphorColorMapper.cc(67): warning C4474: 'sscanf' : too many arguments passed for format string
QFosphorColorMapper.cc(67): note: placeholders and their parameters expect 1 variadic arguments, but 2 were provided
```

A support for `%m` seems Linux specific too me. Ref:  http://linux.die.net/man/3/scanf 

No word on this on [MSDN](https://learn.microsoft.com/en-us/cpp/c-runtime-library/scanf-type-field-characters?view=msvc-170). Not sure if MinGW-w64 supports this. 

For MSVC, that code will crash at runtime; `free()` of an illegal pointer.

I tried this feeble attempt instead:
```diff
--- a/gr-qtgui/lib/QFosphorColorMapper.cc 2023-04-26 15:19:11
+++ b/gr-qtgui/lib/QFosphorColorMapper.cc 2023-05-04 13:47:35
@@ -64,7 +64,13 @@

         /* Are we in a palette ? */
         if (kp < 0) {
+#ifdef _WIN32
+            char buf[41] = { '\0' };
+            name = buf;
+            sscanf(line.constData(), "%d %40s", &kp, buf);
+#else
             sscanf(line.constData(), "%d %ms", &kp, &name);
+#endif

             if (kp)
                 gradient = new QLinearGradient();
@@ -75,8 +81,10 @@

             addPalette(name, pixmap);

+#ifdef _WIN32
             /* Release name & prepare for next */
             free(name);
+#endif
             name = NULL;

             kp = -1;
@@ -96,7 +104,9 @@
                 delete gradient;
                 gradient = NULL;

+#ifdef _WIN32
                 free(name);
+#endif
                 name = NULL;

                 kp = -1;
```

Must be a better way.

### Feature Urgency

high (impacts broad portion of ecosystem)

### More Information

_No response_

--- a/gr-qtgui/lib/QFosphorColorMapper.cc
+++ b/gr-qtgui/lib/QFosphorColorMapper.cc
@@ -9,6 +9,8 @@
 
 #include "QFosphorColorMapper.h"
 #include <QGLContext>
+#include <memory>
+#include <sstream>
 
 namespace gr {
 namespace qtgui {
@@ -45,8 +47,8 @@ QFosphorColorMapper::QFosphorColorMapper(QObject* parent) : QObject(parent)
 
 int QFosphorColorMapper::loadFromFile(QFile& file)
 {
-    QLinearGradient* gradient = NULL;
-    char* name;
+    std::unique_ptr<QLinearGradient> gradient;
+    std::string name;
     int kp = -1;
 
     /* Make sure it's open */
@@ -54,6 +56,7 @@ int QFosphorColorMapper::loadFromFile(QFile& file)
         file.open(QFile::ReadOnly);
 
     /* Scan until the end */
+    std::stringstream ss;
     while (!file.atEnd()) {
         /* Grab a line */
         QByteArray line = file.readLine().simplified();
@@ -64,10 +67,14 @@ int QFosphorColorMapper::loadFromFile(QFile& file)
 
         /* Are we in a palette ? */
         if (kp < 0) {
-            sscanf(line.constData(), "%d %ms", &kp, &name);
+            ss.str(line.toStdString());
+            ss >> kp;
+            ss >> name;
+            ss.clear();
 
-            if (kp)
-                gradient = new QLinearGradient();
+            if (kp) {
+                gradient = std::make_unique<QLinearGradient>();
+            }
         } else if (kp == 0) {
             /* Create palette */
             QString filename = line;
@@ -76,28 +83,25 @@ int QFosphorColorMapper::loadFromFile(QFile& file)
             addPalette(name, pixmap);
 
             /* Release name & prepare for next */
-            free(name);
-            name = NULL;
+            name.clear();
 
             kp = -1;
         } else {
-            float f[4];
+            std::array<float, 4> f;
 
             /* Read data point */
-            sscanf(line.constData(), "%f %f %f %f", f, f + 1, f + 2, f + 3);
-            gradient->setColorAt(f[0], QColor::fromRgbF(f[1], f[2], f[3]));
+            ss.str(line.toStdString());
+            for (auto& val : f) {
+                ss >> val;
+            }
+            ss.clear();
 
+            // gradient should already be initialized in this else clause
+            gradient->setColorAt(f[0], QColor::fromRgbF(f[1], f[2], f[3]));
             /* Is it over ? */
             if (!--kp) {
                 /* Add the newly created palette */
                 addPalette(name, *gradient);
-
-                /* Release gradient, name & prepare for next */
-                delete gradient;
-                gradient = NULL;
-
-                free(name);
-                name = NULL;
 
                 kp = -1;
             }

